python/pyarrow/tests/test_feather.py

```diff
@@ -128,19 +128,22 @@ def test_dataset(version):
     num_values = (100, 100)
```

Comment:
I think the `@pytest.mark.pandas` can be removed here as well?

Reply:
Done

Reply:
For this to work I had to remove the call to `_check_pandas_version` in `FeatherDataset`'s `__init__`. I think that was a bug since `FeatherDataset` is supposed to work without `pandas`. I added a call to `_check_pandas_version` in `FeatherDataset.read_pandas` since we do want it there